src/ngMock/angular-mocks.js

```diff
@@ -1482,25 +1506,48 @@ function createHttpBackendMock($rootScope, $timeout, $delegate, $browser) {
 
     assertArgDefined(arguments, 1, 'url');
 
-    var definition = new MockHttpExpectation(method, url, data, headers, keys),
-        chain = {
-          respond: function(status, data, headers, statusText) {
-            definition.passThrough = undefined;
-            definition.response = createResponse(status, data, headers, statusText);
-            return chain;
-          }
-        };
+    var mock = when(method, url, data, headers, keys);
 
-    if ($browser) {
-      chain.passThrough = function() {
-        definition.response = undefined;
-        definition.passThrough = true;
-        return chain;
-      };
-    }
+    definitions.push(mock.definition);
+    return mock.chain;
+  };
 
-    definitions.push(definition);
-    return chain;
+  /**
+   * @ngdoc method
+   * @name $httpBackend#whenOverride
+   * @description
+   * Creates a new backend definition that will override any already existing definitions with the same parameters
```

Comment:
I think it is important t mention that it will take precendence over all previous definitions (even they have different parameters).

Reply:
Something along these lines:

```
Creates a new backend definition that is put at the front of the queue for matching.
This means that a  matching `whenOverride` definition will always respond even if
there is a more specific `when` definition.
```

And then give some examples such as (thanks to @gkalpak):

**Expected usage:**
```
// Without "override":
$httpBackend.whenPOST('/foo').respond(1);
$httpBackend.whenPOST('/foo', {id: 1}).respond(2);
$http.post('/foo');            // --> 1
$http.post('/foo', {id: 1});   // --> 1

// With "overrride":
$httpBackend.whenPOST('/foo').respond(1);
$httpBackend.whenOverridePOST('/foo', {id: 1}).respond(2);
$http.post('/foo');            // --> 1
$http.post('/foo', {id: 1});   // --> 2
```

**Potentially confusing usage:**
```
// Without "override":
$httpBackend.whenPOST('/foo', {id: 1}).respond(1);
$httpBackend.whenPOST('/foo').respond(2);
$http.post('/foo');            // --> 2
$http.post('/foo', {id: 1});   // --> 1

// With "overrride":
$httpBackend.whenPOST('/foo', {id: 1}).respond(1);
$httpBackend.whenOverridePOST('/foo').respond(2);
$http.post('/foo');            // --> 2
$http.post('/foo', {id: 1});   // --> 2
```